File: VB/CustomActionControl.Module.Win/Controllers/SampleController.vb
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base

Namespace CustomActionControl.Module.Win.Controllers
    Public Class SampleController
        Inherits ObjectViewController(Of ListView, Order)

        Public Sub New()
            Dim activeFilterAction As New CheckableSimpleAction(Me, "ActiveFilterAction", PredefinedCategory.Filters)
            activeFilterAction.Caption = "Active"
            AddHandler activeFilterAction.Execute, AddressOf activeFilterAction_Execute
            Dim todayFilterAction As New CheckableSimpleAction(Me, "TodayFilterAction", PredefinedCategory.Filters)
            todayFilterAction.Caption = "Today"
            AddHandler todayFilterAction.Execute, AddressOf todayFilterAction_Execute
        End Sub
        Private Sub activeFilterAction_Execute(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs)
            Dim applyFilter As Boolean = DirectCast(sender, CheckableSimpleAction).Checked
            If applyFilter Then
                View.CollectionSource.Criteria("IsActive") = New OperandProperty("Active")
            Else
                View.CollectionSource.Criteria.Remove("IsActive")
            End If
        End Sub
        Private Sub todayFilterAction_Execute(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs)
            Dim applyFilter As Boolean = DirectCast(sender, CheckableSimpleAction).Checked
            If applyFilter Then
                View.CollectionSource.Criteria("IsToday") = New FunctionOperator(FunctionOperatorType.IsOutlookIntervalToday, New OperandProperty("Date"))
            Else
                View.CollectionSource.Criteria.Remove("IsToday")
            End If
        End Sub
    End Class
End Namespace

File: CS/EFCore/CustomActionEF/CustomActionEF.Win/Controllers/SampleController.cs
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;

namespace CustomActionControl.Module.Win.Controllers {
    public class SampleController : ObjectViewController<ListView, Order> {
        public SampleController() {
            CheckableSimpleAction activeFilterAction = new CheckableSimpleAction(this, "ActiveFilterAction", PredefinedCategory.Filters);
            activeFilterAction.Caption = "Active";
            activeFilterAction.Execute += activeFilterAction_Execute;
            CheckableSimpleAction todayFilterAction = new CheckableSimpleAction(this, "TodayFilterAction", PredefinedCategory.Filters);
            todayFilterAction.Caption = "Today";
            todayFilterAction.Execute += todayFilterAction_Execute;
        }
        void activeFilterAction_Execute(object sender, DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs e) {
            bool applyFilter = ((CheckableSimpleAction)sender).Checked;
            if (applyFilter){
                View.CollectionSource.Criteria["IsActive"] = new OperandProperty("Active");
            } else {
                View.CollectionSource.Criteria.Remove("IsActive");
            }
        }
        void todayFilterAction_Execute(object sender, DevExpress.ExpressApp.Actions.SimpleActionExecuteEventArgs e) {
            bool applyFilter = ((CheckableSimpleAction)sender).Checked;
            if (applyFilter) {
                View.CollectionSource.Criteria["IsToday"] = new FunctionOperator(FunctionOperatorType.IsOutlookIntervalToday, new OperandProperty("Date"));
            } else {
                View.CollectionSource.Criteria.Remove("IsToday");
            }
        }
    }
}
